src/app/Layouts/defaultPageWrapper.jsx

```diff
@@ -1,51 +1,104 @@
-import React, { Fragment } from 'react';
+import React, { Fragment, useContext } from 'react';
 import { node, string, bool } from 'prop-types';
 import HeaderContainer from '../containers/Header';
 import FooterContainer from '../containers/Footer';
 import ManifestContainer from '../containers/Manifest';
 import ServiceWorkerContainer from '../containers/ServiceWorker';
-import { ServiceContextProvider } from '../contexts/ServiceContext';
+import {
+  ServiceContextProvider,
+  ServiceContext,
+} from '../contexts/ServiceContext';
 import { RequestContextProvider } from '../contexts/RequestContext';
 import ConsentBanner from '../containers/ConsentBanner';
 import getStatsDestination from '../contexts/RequestContext/getStatsDestination';
 import getStatsPageIdentifier from '../contexts/RequestContext/getStatsPageIdentifier';
 import getOriginContext from '../contexts/RequestContext/getOriginContext';
 import getEnv from '../contexts/RequestContext/getEnv';
 import GlobalStyle from '../lib/globalStyles';
+import deepGet from '../lib/utilities/deepGet';
+
+const PageWithRequestContext = ({
+  children,
+  data,
+  env,
+  id,
+  isAmp,
+  isUK,
+  origin,
+  pageType,
+  service,
+}) => {
+  const { lang } = useContext(ServiceContext);
+  const getLangByPageType = type =>
```

Comment:
Can you move this to a helper inside `../contexts/RequestContext` please, this file is getting very polluted IMO. Ive made a PR to refactor https://github.com/bbc/simorgh/pull/2185

Reply:
I disagree with it being moved to `RequestContext`, as this now means the entire `data` object has to be passed through. If anything it should be in `../lib/utilities` and imported here so we don't have to pass the object around.

Reply:
After discussion with @jamesdonoh, the above should be left for a future piece of work.